vortex-datafusion/src/plans.rs

```diff
@@ -182,37 +183,23 @@ impl Stream for RowIndicesStream {
             .project(this.filter_projection.as_slice())
             .expect("projection should succeed");
 
-        // Immediately convert to Arrow RecordBatch for processing.
-        // TODO(aduffy): attempt to pushdown the filter to Vortex without decoding.
-        let record_batch = RecordBatch::from(
-            vortex_struct
-                .into_canonical()
-                .unwrap()
-                .into_arrow()
-                .as_struct(),
-        );
-
-        // Generate a physical plan to execute the conjunction query against the filter columns.
-        //
-        // The result of a conjunction expression is a BooleanArray containing `true` for rows
-        // where the conjunction was satisfied, and `false` otherwise.
-        let df_schema = DFSchema::try_from(this.filter_schema.clone())?;
-        let physical_expr =
-            create_physical_expr(&this.conjunction_expr, &df_schema, &Default::default())?;
-        let selection = physical_expr
-            .evaluate(&record_batch)?
-            .into_array(record_batch.num_rows())?;
+        // TODO(adamg): Filter on vortex arrays
+        let array = ExpressionEvaluator::eval(vortex_struct, &this.conjunction_expr).unwrap();
+        let selection = array
+            .into_bool()
+            .unwrap()
+            .into_canonical()
+            .unwrap()
+            .into_arrow();
```

Comment:
This is actually just 
```suggestion
        let selection = array.into_canonical().unwrap().into_arrow();
```
since everything returns an array you can get lost as to what has been converted already. FWIW `into_bool()` is a shortcut for `into_canonical().into_bool()`